### ISSUE TITLE
Can the Data Flow Graph in mediapipe be separated and become a third-party library?

### ISSUE THREAD

**Issue by libaineu2004** [google/mediapipe#4184]
### MediaPipe Solution (you are using)

Data Flow Graph

### Programming language

C++

### Are you willing to contribute it

No

### Describe the feature and the current behaviour/state

```shell
Can the Data Flow Graph in mediapipe be separated and become a third-party library? Use C++ language.
Like this open source project:
https://github.com/taskflow/taskflow

Purely a library, just a function of the flowchart.

There are too many features in mediapipe, and I just need flow-chart functionality.
```

### Please specify the use cases for this feature

```shell
null
```

**Comment by ayushgdev** [comment]
Hello @libaineu2004 
This is an interesting ask. While we are delighted to know MediaPipe's graph based implementation is useful for users in this realm as well, MediaPipe is intended to be a customisable ML solution API and not graph based task decomposition for any general use case. We use the graph structure since it best suits the ML solutions need, however, we do not plan to segregate this from the solutions.

**Comment by libaineu2004** [comment]
> Hello @libaineu2004 This is an interesting ask. While we are delighted to know MediaPipe's graph based implementation is useful for users in this realm as well, MediaPipe is intended to be a customisable ML solution API and not graph based task decomposition for any general use case. We use the graph structure since it best suits the ML solutions need, however, we do not plan to segregate this from the solutions.

I'm sorry to hear that. Because this project is too difficult to compile and pass. I submitted the issue, https://github.com/google/mediapipe/issues/4171. I really only want to use the flowchart function.

**Comment by ayushgdev** [comment]
@libaineu2004 We understand it might be difficult to compile the project. Hence we are working on new MediaPipe Tasks API which can be much more easier to use and resilient to user-defined configurations. 
However, we will not be able to segregate the graph utilities as a separate library for Data Flow.

**Comment by libaineu2004** [comment]
> MediaPipe

Yes, it's hard to compile. If you can directly provide vs2019 version of MediaPipe *.dll/*.lib/*,h is also a good choice.

**Comment by schmidt-sebastian** [comment]
We currently do not have plans to support this but can take external contirbutions.
